snuba/datasets/events_processor.py

```diff
@@ -0,0 +1,453 @@
+
+    def _extract_http(self, output, http):
+        output['http_method'] = _unicodify(http.get('method', None))
+        http_headers = _as_dict_safe(http.get('headers', None))
+        output['http_referer'] = _unicodify(http_headers.get('Referer', None))
+
+    def _extract_stacktraces(self, output, stacks):
+        stack_types = []
+        stack_values = []
+        stack_mechanism_types = []
+        stack_mechanism_handled = []
+
+        frame_abs_paths = []
+        frame_filenames = []
+        frame_packages = []
+        frame_modules = []
+        frame_functions = []
+        frame_in_app = []
+        frame_colnos = []
+        frame_linenos = []
+        frame_stack_levels = []
+
+        stack_level = 0
+        for stack in stacks:
+            if stack is None:
+                continue
+
+            stack_types.append(_unicodify(stack.get('type', None)))
+            stack_values.append(_unicodify(stack.get('value', None)))
+
+            mechanism = stack.get('mechanism', None) or {}
+            stack_mechanism_types.append(_unicodify(mechanism.get('type', None)))
+            stack_mechanism_handled.append(_boolify(mechanism.get('handled', None)))
+
+            frames = (stack.get('stacktrace', None) or {}).get('frames', None) or []
+            for frame in frames:
+                if frame is None:
+                    continue
+
+                frame_abs_paths.append(_unicodify(frame.get('abs_path', None)))
+                frame_filenames.append(_unicodify(frame.get('filename', None)))
+                frame_packages.append(_unicodify(frame.get('package', None)))
+                frame_modules.append(_unicodify(frame.get('module', None)))
+                frame_functions.append(_unicodify(frame.get('function', None)))
+                frame_in_app.append(frame.get('in_app', None))
+                frame_colnos.append(_collapse_uint32(frame.get('colno', None)))
+                frame_linenos.append(_collapse_uint32(frame.get('lineno', None)))
+                frame_stack_levels.append(stack_level)
+
+            stack_level += 1
+
+        output['exception_stacks.type'] = stack_types
+        output['exception_stacks.value'] = stack_values
+        output['exception_stacks.mechanism_type'] = stack_mechanism_types
+        output['exception_stacks.mechanism_handled'] = stack_mechanism_handled
+        output['exception_frames.abs_path'] = frame_abs_paths
+        output['exception_frames.filename'] = frame_filenames
+        output['exception_frames.package'] = frame_packages
+        output['exception_frames.module'] = frame_modules
+        output['exception_frames.function'] = frame_functions
+        output['exception_frames.in_app'] = frame_in_app
+        output['exception_frames.colno'] = frame_colnos
+        output['exception_frames.lineno'] = frame_linenos
+        output['exception_frames.stack_level'] = frame_stack_levels
+
+
+class EventTooOld(Exception):
+    pass
+
+
+def _as_dict_safe(value):
```

Comment:
The methods here and below are likely useful for others processors as well. I would suggest you leave them where they were (I was planning to use unicodify for the groupedmessage for  example, but it turned out I did not need strings.